ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/validation/BlockValidator.java

```diff
@@ -101,19 +102,36 @@ public BlockValidator(final Spec spec, RecentChainData recentChainData) {
                           parentBlock.get().getSlot().max(firstSlotInBlockEpoch),
                           block.getParentRoot()))
                   .thenApply(
-                      postState -> {
-                        if (postState.isEmpty()) {
+                      maybePostState -> {
+                        if (maybePostState.isEmpty()) {
                           LOG.trace(
                               "Block was available but state wasn't. Must have been pruned by finalized.");
                           return InternalValidationResult.IGNORE;
                         }
-                        if (!blockIsProposedByTheExpectedProposer(block, postState.get())) {
+                        final BeaconState postState = maybePostState.get();
+
+                        if (!blockIsProposedByTheExpectedProposer(block, postState)) {
                           return reject(
                               "Block proposed by incorrect proposer (%s)",
                               block.getProposerIndex());
                         }
-                        if (!blockSignatureIsValidWithRespectToProposerIndex(
-                            block, postState.get())) {
+                        if (spec.atSlot(block.getSlot()).miscHelpers().isMergeComplete(postState)) {
+                          ExecutionPayload executionPayload =
+                              block
+                                  .getMessage()
+                                  .getBody()
+                                  .getOptionalExecutionPayload()
+                                  .orElseThrow();
```

Comment:
nit: I think this is right because to parse the block we'd have to get a payload and we're just checking if it's the default payload or not.  But I think it would be clearer if instead of `.orElseThrow` we just returned `reject("Missing execution payload")` if it's not present.

Reply:
yeah, will never happen but going to change it